src/golang/lib/workflow/utils/database.go

```diff
@@ -374,6 +376,57 @@ func UpdateDAGResultMetadata(
 		return err
 	}
 
+	// Check if DAG has finished running.
+	if execState.Status == shared.CanceledExecutionStatus || execState.Status == shared.FailedExecutionStatus || execState.Status == shared.SucceededExecutionStatus {
```

Comment:
If the DAG status is succeeded, do we need to perform the following checks? The artifact result and op results should all be succeeded already?